File: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-10-01-preview/examples-java/Namespaces/ListBySubscription.java

/**
 * Samples for Namespaces List.
 */
public final class Main {
    /*
     * x-ms-original-file:
     * specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-10-01-preview/examples/
     * Namespaces/ListBySubscription.json
     */
    /**
     * Sample code: Namespaces_ListAll.
     * 
     * @param manager Entry point to NotificationHubsManager.
     */
    public static void namespacesListAll(com.azure.resourcemanager.notificationhubs.NotificationHubsManager manager) {
        manager.namespaces().list(null, null, com.azure.core.util.Context.NONE);
    }
}

File: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-10-01-preview/examples-js/Namespaces/ListBySubscription.js
const { NotificationHubsManagementClient } = require("@azure/arm-notificationhubs");
const { DefaultAzureCredential } = require("@azure/identity");

/**
 * This sample demonstrates how to Lists all the available namespaces within the subscription.
 *
 * @summary Lists all the available namespaces within the subscription.
 * x-ms-original-file: specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/preview/2023-10-01-preview/examples/Namespaces/ListBySubscription.json
 */
async function namespacesListAll() {
  const subscriptionId =
    process.env["NOTIFICATIONHUBS_SUBSCRIPTION_ID"] || "29cfa613-cbbc-4512-b1d6-1b3a92c7fa40";
  const credential = new DefaultAzureCredential();
  const client = new NotificationHubsManagementClient(credential, subscriptionId);
  const resArray = new Array();
  for await (let item of client.namespaces.listAll()) {
    resArray.push(item);
  }
  console.log(resArray);
}
